src/t_string.c

```diff
@@ -75,11 +75,11 @@ void setGenericCommand(client *c, int flags, robj *key, robj *val, robj *expire,
     if (expire) {
         if (getLongLongFromObjectOrReply(c, expire, &milliseconds, NULL) != C_OK)
             return;
+        if (unit == UNIT_SECONDS) milliseconds *= 1000;
```

Comment:
this can back fire too.
if the input was negative, multiplying it by 1000 can make it positive.
i think the intent of verifying the input may be enough, but if we want to a hermetic check, we probably need a range check, or two negative checks.

Reply:
I guess two negative checks are fine. wdyt?

Reply:
Added additional negative check for overflow detection. Here's the result:
```
127.0.0.1:6379> SET A B EX -9999999999999999
(error) ERR invalid expire time in set
127.0.0.1:6379> SET A B EX 10000000000000000
(error) ERR invalid expire time in set
```